packages/client/lib/index.js

```diff
@@ -1,15 +1,71 @@
+const shuffle = require('lodash.shuffle')
 const HttpClient = require('./http')
 const resources = require('./resources')
+const initialPeers = require('./peers')
+const sortPeers = require('./utils/sort-peers')
 
 module.exports = class ApiClient {
+  /**
+   * Finds all the available peers, sorted by block heigh and delay
+   *
+   * @param {String} network - Network name
+   * @param {Number} version - API version
+   */
+  static async findPeers (network, version) {
```

Comment:
This really shouldn't be the job of an API client, the developer that implements the API client should implement a way of finding and filtering peers he would like to use.

All `*-client` packages should only manage the interaction with the API **(send request, return response)** and not implement anything that is implementation specific to other applications.

This looks more like a utility for wallets then something that should be part of any `client` packages.

Reply:
i disagree, this is something a bit more subtle:
- either you have a full node you ping. This full node is making some work to keep in sync, connecting to random nodes, as random as possible to prevent from attacks. you trust YOUR node
- as a lite client, you are trusting someone else node. To prevents attacks, this is up to the client to be able to play the random access game and find correct peers by analysing the network

Reply:
I understand your idea, but I find more practical having something like this.

I mean, with the typical API client you don't need to know which host is online. You just make requests. When you're just "playing" with the client, it should be straightforward, it shouldn't be very involved.

I started to implement this for the new wallet, but I realized that we've this in different places (the current `desktop-wallet` and the mobile-wallet`, which reuses part of `ark-ts`, which uses parts of `ark-js`) and that, for more use cases, would be useful:

For instance, if a user wants to get some of her transactions to do her taxes, we shouldn't force her to find out how to get which hosts are available for `mainnet` (maybe, she would search that on Google, on GitHub, on the wallet, maybe she would ask on Slack, etc.).

Maybe 1 month later, she try her script/app and it doesn't work because the node is down. So she has to search for a new peer again.

Reply:
I'm with @faustbrian on this one, clients shouldn't be that smart. They should be light and "dumb" and only offer a way to communicate with an API.

Everything else (eg. fetching peers, getting the peer with least latency etc.) should be up to the app to decide what it wants to do and how. If you want to have more complex logic create a separate module for it or if its a single function, put it into utils for example

Reply:
I believe @fix and @j-a-m-l are on your side.

Reply:
To me, the main problem is that, if we don't add this, @roks0n, we have a half-baked solution that implies that every developer that wants to use this client, should choose between using hardcoded peers or implement something like this. And that is not user friendly at all.